test/integration/consul-container/test/README.md

```diff
@@ -0,0 +1,67 @@
+# Upgrade Integration Tests
+
+The upgrade integration test is responsible for ensuring that users are able to upgrade between consul versions without experiencing difficulties. The upgrade tests does the following:
+- Ensure that customers are able to upgrade to a new Consul version and new and existing features work as expected. 
+- Test for backwards compatibility. i.e support for upgrading directly from Consul `1.13` to `1.15`, without upgrading to `1.14`
+    - Highlights errors that will occur if attempting to skip a version during upgrades. Example the API changed between version 1.13 and 1.15 so direct upgrades between these version is not supported. 
+
+
```

Comment:
In this introductory section, we could briefly explains how the end to end upgrade test works, like follows

```markdown
Following is the main steps of an upgrade test case:

1. Create a cluster with a specified number of server and client agents, and enable the feature to be tested.
2. Create some workloads in the cluster, e.g., register 2 services: static-server, static-client.
3. Verify connection / disconnection (e.g., deny Action)
4. Upgrade consul cluster and restart the envoy sidecars (we restart envoy sidecar to ensure the upgraded consul binary can read the state from the previous version and generate the correct envoy configurations)

5. Verify connection / disconnection (e.g., deny Action)

```